### PR TITLE
Enable annotations for service account in fluentbit/fluentd

### LINKED ISSUE #562: Support Dynamic Reload when an annotation is added to a service account
### Is your feature request related to a problem? Please describe.

Using fluentbit, I am trying to get Cloudwatch to work as an output plugin. See below for a very basic configuration setup:

```
apiVersion: fluentbit.fluent.io/v1alpha2
kind: ClusterInput
metadata:
  labels:
    fluentbit.fluent.io/component: logging
    fluentbit.fluent.io/enabled: "true"
  name: tail
spec:
  tail:
    db: /fluent-bit/tail/pos.db
    dbSync: Normal
    memBufLimit: 5MB
    parser: docker
    path: /var/log/containers/*.log
    refreshIntervalSeconds: 10
    skipLongLines: true
    tag: kube.*
---
apiVersion: fluentbit.fluent.io/v1alpha2
kind: ClusterOutput
metadata:
  namespace: fluent
  name: cloudwatch-output
  labels:
    fluentbit.fluent.io/enabled: "true"
    fluentbit.fluent.io/mode: "fluentbit-only"
spec:
  customPlugin:
    config: |
      Name       cloudwatch_logs
      Match      kube.*
      region     <region>
      log_group_name  <name>
      log_stream_name <name>
```

I am also following option 3 (3. EKS Web Identity Token (OIDC)) as a method of authenticating to AWS as per the fluentbit documentation [here](https://github.com/fluent/fluent-bit-docs/blob/43c4fe134611da471e706b0edb2f9acd7cdfdbc3/administration/aws-credentials.md#3-eks-web-identity-token-oidc). This involves adding an annotation to the fluent-bit service account. 

When this annotation is added, I'd expect the fluentbit pods to be able to authenticate with Cloudwatch. However, I must first delete the fluentbit pod before the authentication and output plugin works as expected.

### Describe the solution you'd like

I'm unsure on if this will work, but does changing the `FluentBitSpec` cause the app to dynamically reload? If so, then setting some field such called `serviceAccountAnnotations` [here](https://github.com/fluent/fluent-operator/blob/master/docs/fluentbit.md#fluentbitspec) would solve the issue.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by HarleyB123**
### What this PR does / why we need it:

Currently, if you wanted to use Workload Identity - you'd have to manually annotate the fluentbit/fluentd service account, and then restart the pods for authentication to occur. (Sometimes it worked without authentication). This PR allows for annotations to be added to the service account and a restart of fluentbit/fluentd to occur.

### Which issue(s) this PR fixes:

Fixes #562 

### Does this PR introduced a user-facing change?

```release-note
None (or not really, more an optional field than any action required)
```

### Additional documentation, usage docs, etc.:

```docs
See PR for docs.md
```

### CODE REVIEW

**Comment by benjaminhuo** [comment]
@HarleyB123 You may need to fix the CI error as well :)

**Comment by benjaminhuo** [comment]
> > @HarleyB123 You may need to fix the CI error as well :)
> 
> All I can see from CI errors is:
> 
> > GitHub Actions has encountered an internal error when running your job.
> 
> Looks like it's an [issue](https://github.com/actions/runner-images/issues/2179) at the moment, can retry later

Ops, github is still not stable after 24 hours...

File: pkg/operator/rbac.go [hunk 1]
```diff
@@ -29,8 +29,9 @@ func MakeRBACObjects(name, namespace, component string, additionalRules []rbacv1
 
 	sa := corev1.ServiceAccount{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: namespace,
+			Name:        name,
+			Namespace:   namespace,
+			Annotations: fbAnnotations,
```
Comment (wenchajun) [inline pkg/operator/rbac.go:34]: If the user does not set the `serviceAccountAnnotations` field, then this field will be empty, is this case considered?

Comment (benjaminhuo) [reply pkg/operator/rbac.go:34]: It's ok to set an empty annotation, right? @wanjunlei

File: pkg/operator/rbac.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func MakeRBACObjects(name, namespace, component string, additionalRules []rbacv1.PolicyRule) (*rbacv1.ClusterRole, *corev1.ServiceAccount, *rbacv1.ClusterRoleBinding) {
+func MakeRBACObjects(name, namespace, component string, additionalRules []rbacv1.PolicyRule, fbAnnotations map[string]string) (*rbacv1.ClusterRole, *corev1.ServiceAccount, *rbacv1.ClusterRoleBinding) {
```
Comment (benjaminhuo) [inline pkg/operator/rbac.go:11]: MakeRBACObjects is also used by fluentd, so maybe we can change `fbAnnotations` to `saAnnotations`

File: pkg/operator/rbac.go [hunk 3]
```diff
@@ -55,7 +56,7 @@ func MakeRBACObjects(name, namespace, component string, additionalRules []rbacv1
 	return &cr, &sa, &crb
 }
 
-func MakeScopedRBACObjects(fbName, fbNamespace string) (*rbacv1.Role, *corev1.ServiceAccount, *rbacv1.RoleBinding) {
+func MakeScopedRBACObjects(fbName, fbNamespace string, fbAnnotations map[string]string) (*rbacv1.Role, *corev1.ServiceAccount, *rbacv1.RoleBinding) {
```
Comment (benjaminhuo) [inline pkg/operator/rbac.go:59]: saAnnotations too here?

File: pkg/operator/rbac.go (hunk 1 above)
Comment (wanjunlei) [reply pkg/operator/rbac.go:34]: Yes

File: pkg/operator/rbac.go (hunk 2 above)
Comment (HarleyB123) [reply pkg/operator/rbac.go:11]: Ah good spot! will change

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@HarleyB123 Have invited you as fluent operator maintainer, and also proposed to add you as fluent org member

**Pull request merged**